clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -177,7 +177,7 @@ private void process(final FetchCommittedOffsetsEvent event) {
             return;
         }
         CommitRequestManager manager = requestManagers.commitRequestManager.get();
-        long expirationTimeMs = getExpirationTimeForTimeout(event.timeout());
+        long expirationTimeMs = getExpirationTimeForTimeout(event.deadlineMs());
```

Comment:
Isn't `event.deadlineMs()` already a point in time instead of a time span? If yes, you do not need to compute the point in time anymore, do you?

Reply:
Correct. This PR is intentionally limited in its design and scope. My hope was to leave that for this PR and rectify it as part of KAFKA-15974.

Reply:
Wait, I think we are misunderstanding each other. Method `getExpirationTimeForTimeout()` computes a point in time by adding a given timeout, i.e., a time delta, to the the current time. Method `event.deadlineMS()` returns a point in time computed by adding the remaining time of the timer, i.e., a time delta, to the current time. With this call you are trying to compute a point in time by adding a time point to the current time. This does not look like a limitation but rather like a mistake.

Reply:
You're right! Thanks for catching that!

Previously, we used delta values and we're moving to absolute values. The absolute value is now calculated once in the `CompletableApplicationEvent` and used subsequently.